homeassistant/components/netatmo.py

```diff
@@ -50,7 +64,10 @@ def setup(hass, config):
             "Please check your settings for NatAtmo API.")
         return False
 
-    for component in 'camera', 'sensor':
+    components = [component for component in ('camera', 'sensor')
```

Comment:
Ok, sorry then.

Then the easiest way to achieve that is with a try-except when doing the discovery because if there is no camera, in its current status, the lnetatmo module raises the error refered in #2601 .

Or do you prefer to perform the check in the /components/netatmo.py and then launch the discovery if the test is successful?